### ISSUE TITLE
Docker source results in warnings for parsing label.com.docker.swarm.task

### ISSUE THREAD

**Issue by AlexGustafsson** [vectordotdev/vector#3305]
First off, I'm unsure of how to best describe this issue, so there's a lot of logs attached below.

Using the Docker source, it seems as if there's some bug or edge case where the `label.com.docker.swarm.task` field will be incorrectly parsed. This seems to trigger a retry which is then rate limited. All in all I get more or less constant logs from all vector nodes. 

The sink I'm using is Elasticsearch. I seem to get no errors on that side (INFO and above).

```
Aug 01 08:28:16.962  INFO vector::sources::docker: Started listening logs on docker container id=7020eac10248e76c6369a65eef120d31d8f776d0a7f110ae6ca9971496f5d75e
...
Aug 01 08:28:17.076  INFO vector::topology::builder: Healthcheck: Passed.
Aug 01 08:28:24.562  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=0}: vector::sinks::elasticsearch: ElasticSearch error response err_type=mapper_parsing_exception reason=object mapping for [label.com.docker.swarm.task] tried to parse field [task] as object, but found a concrete value rate_limit_secs=30
Aug 01 08:28:24.662  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=0}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:28:25.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=1}: vector::sinks::elasticsearch: "ElasticSearch error response" is being rate limited. rate_limit_secs=5
Aug 01 08:28:25.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=1}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:28:39.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=2}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:28:55.062  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=3}: vector::sinks::elasticsearch: 3 "ElasticSearch error response" events were rate limited. rate_limit_secs=5
Aug 01 08:28:55.062  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=3}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:29:09.868  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=4}: vector::sinks::elasticsearch: ElasticSearch error response err_type=mapper_parsing_exception reason=object mapping for [label.com.docker.swarm.task] tried to parse field [task] as object, but found a concrete value rate_limit_secs=30
Aug 01 08:29:09.869  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=4}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:29:24.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=5}: vector::sinks::elasticsearch: "ElasticSearch error response" is being rate limited. rate_limit_secs=5
Aug 01 08:29:24.962  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=5}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:29:40.162  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=6}: vector::sinks::elasticsearch: 2 "ElasticSearch error response" events were rate limited. rate_limit_secs=5
Aug 01 08:29:40.163  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=6}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:29:52.663  INFO vector::sources::docker: Stoped listening logs on docker container id=62b279733fab72a733cb11cb2321e8069bce1034cf2905073de241e3a3ca0fbb
Aug 01 08:29:53.562  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=7}: vector::sinks::elasticsearch: ElasticSearch error response err_type=mapper_parsing_exception reason=failed to parse field [label.com.docker.swarm.task] of type [text] in document with id 'XokkqXMBbxYl9oArRXW_'. Preview of field's value: '{name=monitoring_cadvisor.84yv6zw0olgqmw51omuevrirn.8vbzo69ks9ypcxu24l89xup3h}' rate_limit_secs=30
Aug 01 08:29:53.562  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=7}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:29:54.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=8}: vector::sinks::elasticsearch: "ElasticSearch error response" is being rate limited. rate_limit_secs=5
Aug 01 08:29:54.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=8}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:30:01.168  INFO vector::sources::docker: Started listening logs on docker container id=8c2352346a3eb9a08bc16bf92a6af2fae1fd11b12f936c9bee4304de42746d21
Aug 01 08:30:02.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=9}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:30:09.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=10}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:30:24.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=11}: vector::sinks::elasticsearch: 4 "ElasticSearch error response" events were rate limited. rate_limit_secs=5
Aug 01 08:30:24.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=11}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:30:39.663  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=12}: vector::sinks::elasticsearch: ElasticSearch error response err_type=mapper_parsing_exception reason=object mapping for [label.com.docker.swarm.task] tried to parse field [task] as object, but found a concrete value rate_limit_secs=30
Aug 01 08:30:39.663  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=12}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:30:40.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=13}: vector::sinks::elasticsearch: "ElasticSearch error response" is being rate limited. rate_limit_secs=5
Aug 01 08:30:40.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=13}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:30:55.062  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=14}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:31:10.062  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=15}: vector::sinks::elasticsearch: 3 "ElasticSearch error response" events were rate limited. rate_limit_secs=5
Aug 01 08:31:10.062  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=15}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:31:24.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=16}: vector::sinks::elasticsearch: ElasticSearch error response err_type=mapper_parsing_exception reason=object mapping for [label.com.docker.swarm.task] tried to parse field [task] as object, but found a concrete value rate_limit_secs=30
Aug 01 08:31:24.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=16}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:31:25.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=17}: vector::sinks::elasticsearch: "ElasticSearch error response" is being rate limited. rate_limit_secs=5
Aug 01 08:31:25.863  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=17}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:31:39.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=18}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:31:42.566  INFO vector::sources::docker: Stoped listening logs on docker container id=8c2352346a3eb9a08bc16bf92a6af2fae1fd11b12f936c9bee4304de42746d21
Aug 01 08:31:43.262  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=19}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:31:50.965  INFO vector::sources::docker: Started listening logs on docker container id=4b496de80908ef7d5a3f617455ac5bb47e19187414495a34ebe124cb1ecf061c
Aug 01 08:31:52.962  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=21}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:31:54.763  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=22}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:31:55.778  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=23}: vector::sinks::elasticsearch: 6 "ElasticSearch error response" events were rate limited. rate_limit_secs=5
Aug 01 08:31:55.778  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=23}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:32:09.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=24}: vector::sinks::elasticsearch: ElasticSearch error response err_type=mapper_parsing_exception reason=object mapping for [label.com.docker.swarm.task] tried to parse field [task] as object, but found a concrete value rate_limit_secs=30
Aug 01 08:32:09.962  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=24}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:32:24.462  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=25}: vector::sinks::elasticsearch: "ElasticSearch error response" is being rate limited. rate_limit_secs=5
Aug 01 08:32:24.462  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=25}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:32:25.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=26}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:32:39.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=27}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:32:54.962  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=28}: vector::sinks::elasticsearch: 4 "ElasticSearch error response" events were rate limited. rate_limit_secs=5
Aug 01 08:32:54.962  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=28}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:33:09.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=29}: vector::sinks::elasticsearch: ElasticSearch error response err_type=mapper_parsing_exception reason=object mapping for [label.com.docker.swarm.task] tried to parse field [task] as object, but found a concrete value rate_limit_secs=30
Aug 01 08:33:09.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=29}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:33:24.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=30}: vector::sinks::elasticsearch: "ElasticSearch error response" is being rate limited. rate_limit_secs=5
Aug 01 08:33:24.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=30}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:33:25.862  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=31}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
Aug 01 08:33:39.762  WARN sink{name=elasticsearch type=elasticsearch}:request{request_id=32}: vector::sinks::util::retries2: request is not retryable; dropping the request. reason=some messages failed
```

**Comment by binarylogic** [comment]
Hi @AlexGustafsson, thanks for reporting. Just to clarify a few things:

1. What value are you expecting for `label.com.docker.swarm.task`? It looks like Elasticsearch is expecting an "object" but it appears that it's a scalar.
2. Which version of Vector are you using?

Thanks.

**Comment by AlexGustafsson** [comment]
1. I'm not expecting anything per say, but from the logs it seems as if the Docker source uses `label.com.docker.swarm.task` as an object, but via `docker inspect --format='{{json .Config.Labels}}' <container id>` one can see that the labels are all flat:

```json
{
  "com.docker.stack.namespace": "logs",
  "com.docker.swarm.node.id": "sn7f5zs9544cao1dsap272uq8",
  "com.docker.swarm.service.id": "qg9nae72knvfet28pojc5bmt5",
  "com.docker.swarm.service.name": "logs_vector",
  "com.docker.swarm.task": "",
  "com.docker.swarm.task.id": "e9my36332lacbhhv2jofy6axk",
  "com.docker.swarm.task.name": "logs_vector.sn7f5zs9544cao1dsap272uq8.e9my36332lacbhhv2jofy6axk"
}
```

So I guess I'm either expecting `label.com.docker.swarm.task` to be an empty string, as in the example above, or not to be there at all - since it seems as if it's only there to contain sub-labels. I suppose I expect the sub-labels `com.docker.swarm.task.id` and `com.docker.swarm.task.name` to stand on their own. If I look through what's saved in the Elasticsearch cluster, I can see that indeed `label.com.docker.swarm.task.id` and `label.com.docker.swarm.task.name` exist as fields, but  `label.com.docker.swarm.task` does not - so for the most part, it seems to be already working as I'd expect.

I'm not sure if it changes anything, but I've not done any modifications to any mapping etc., solely relying on Vector for creating the indices and elements. The issue persists if indices are removed in order for them to be created by Vector from scratch.

2. The version I'm using is 0.10.0 via the Docker image `timberio/vector:0.10.0-debian` (`sha256:e6299b59abc776d85bd9a98ba19b8e47401f527f57c858801763af4da65257db`).

**Comment by AlexGustafsson** [comment]
I just checked the Kibana index pattern I use for all of my collected logs. It states that the `label.com.docker.swarm.task` results in a field type conflict since it does not match the expected types. It's about a 50/50 split between containers specifying `text` and `object`. Examples docker images with the `text` type are: vector, swarmpit, traefik, cadvisor, elasticsearch etc. Example images with the `object` type are: Wordpress, gitlab runners, Grafana, prometheus, InfluxDB and traefik. As you can tell there seems to be some large inconsistencies with how this label is handled. It's the only collision among 200+ fields.

**Comment by pgassmann** [comment]
Same issue as #5716

Vector should not try to parse labels with dots as nested structure!

**Comment by otisg** [comment]
FWIW, we had to apply a workaround for one of our users that, we believe, would not have been needed if the fix for this issue and #5716 were in Vector.  @rabbitstack can share more info if needed.

**Comment by jszwedko** [comment]
This should have been fixed by #7152 (this issue is the same as #5716) as noted. Please let me know if you are still seeing this issue though.
